modules/base/rendering/grids/renderablegrid.cpp

```diff
@@ -0,0 +1,320 @@
+            );
+        }
+    );
+
+    glGenVertexArrays(1, &_vaoID);
+    glGenBuffers(1, &_vBufferID);
+
+    glBindVertexArray(_vaoID);
+    glBindBuffer(GL_ARRAY_BUFFER, _vBufferID);
+    glEnableVertexAttribArray(0);
+    glBindVertexArray(0);
+}
+
+void RenderableGrid::deinitializeGL() {
+    glDeleteVertexArrays(1, &_vaoID);
+    _vaoID = 0;
+
+    glDeleteBuffers(1, &_vBufferID);
+    _vBufferID = 0;
+
+    BaseModule::ProgramObjectManager.release(
+        ProgramName,
+        [](ghoul::opengl::ProgramObject* p) {
+            global::renderEngine.removeRenderProgram(p);
+        }
+    );
+    _gridProgram = nullptr;
+}
+
+void RenderableGrid::render(const RenderData& data, RendererTasks&){
+    _gridProgram->activate();
+
+    _gridProgram->setUniform("opacity", _opacity);
+
+    glm::dmat4 modelTransform =
+        glm::translate(glm::dmat4(1.0), data.modelTransform.translation) * // Translation
+        glm::dmat4(data.modelTransform.rotation) *  // Spice rotation
+        glm::scale(glm::dmat4(1.0), glm::dvec3(data.modelTransform.scale));
+
+    glm::dmat4 modelViewTransform = data.camera.combinedViewMatrix() * modelTransform;
+
+    _gridProgram->setUniform("modelViewTransform", modelViewTransform);
+    _gridProgram->setUniform(
+        "MVPTransform",
+        glm::dmat4(data.camera.projectionMatrix()) * modelViewTransform
+    );
+
+    _gridProgram->setUniform("gridColor", _gridColor);
+
+    // Saves current state:
+    GLboolean isBlendEnabled = glIsEnabledi(GL_BLEND, 0);
+    GLfloat currentLineWidth;
+    glGetFloatv(GL_LINE_WIDTH, &currentLineWidth);
+    GLboolean isLineSmoothEnabled = glIsEnabled(GL_LINE_SMOOTH);
+
+    GLenum blendEquationRGB;
+    GLenum blendEquationAlpha;
+    GLenum blendDestAlpha;
+    GLenum blendDestRGB;
+    GLenum blendSrcAlpha;
+    GLenum blendSrcRGB;
+    glGetIntegerv(GL_BLEND_EQUATION_RGB, &blendEquationRGB);
+    glGetIntegerv(GL_BLEND_EQUATION_ALPHA, &blendEquationAlpha);
+    glGetIntegerv(GL_BLEND_DST_ALPHA, &blendDestAlpha);
+    glGetIntegerv(GL_BLEND_DST_RGB, &blendDestRGB);
+    glGetIntegerv(GL_BLEND_SRC_ALPHA, &blendSrcAlpha);
+    glGetIntegerv(GL_BLEND_SRC_RGB, &blendSrcRGB);
+
+    // Changes GL state:
+    glLineWidth(_lineWidth);
+    glEnablei(GL_BLEND, 0);
+    glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+    glEnable(GL_LINE_SMOOTH);
+
+    glBindVertexArray(_vaoID);
+    glDrawArrays(_mode, 0, static_cast<GLsizei>(_varray.size()));
+    glBindVertexArray(0);
+
+    _gridProgram->deactivate();
+
+    // Restores GL State
+    glLineWidth(currentLineWidth);
+    glBlendEquationSeparate(blendEquationRGB, blendEquationAlpha);
+    glBlendFuncSeparate(blendSrcRGB, blendDestRGB, blendSrcAlpha, blendDestAlpha);
+
+    if (!isBlendEnabled) {
+        glDisablei(GL_BLEND, 0);
+    }
+
+    if (!isLineSmoothEnabled) {
+        glDisable(GL_LINE_SMOOTH);
+    }
+}
+
+void RenderableGrid::update(const UpdateData&) {
+    if (_gridIsDirty) {
```

Comment:
To reduce the number of whitespaces (always a good idea), i'd suggest to write it as:

```
if (!_gridIsDirty) {
  return;
}

// ...
```

That way you don't have all of the function inlined